### ISSUE TITLE
streamBuilder vs Consumer / changeNotifier

### ISSUE THREAD

**Issue by ninest** [bizz84/firebase_user_avatar_flutter#2]
For managing `authState`, why do you use a `streamBuilder`? Is not not suitable to use a `Consumer()`?

I just thought that that if you are using Provider, there is no need to have `streamBuilder` as the parent widget. Am I missing something? Can the `authState` widget be made **without** `streamBuilder`?

**Comment by talatkuyuk** [comment]
Yes, you can. I have implemented authbuilderwidget with StreamProvider. It works well.

**Comment by mdev88** [comment]
I am struggling with this exact thing, plus Provider 5.0.0 changed some implementations and broke things. Can anyone provide some up to date code for this app?

Thanks!
